clients/src/test/java/org/apache/kafka/common/UuidTest.java

```diff
@@ -21,50 +21,50 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 
-public class UUIDTest {
+public class UuidTest {
 
     @Test
     public void testSignificantBits() {
-        UUID id = new UUID(34L, 98L);
+        Uuid id = new Uuid(34L, 98L);
 
         assertEquals(id.getMostSignificantBits(), 34L);
         assertEquals(id.getLeastSignificantBits(), 98L);
     }
 
     @Test
-    public void testUUIDEquality() {
-        UUID id1 = new UUID(12L, 13L);
-        UUID id2 = new UUID(12L, 13L);
-        UUID id3 = new UUID(24L, 38L);
+    public void testUuidEquality() {
```

Comment:
Can we add a test that verifies that the `hashCode` is the same for our `Uuid` and Java's `UUID`? Or do we have that already?

Reply:
We don't have that yet. I also didn't specify this behavior. Should I?

Reply:
We don't have to specify it, but it would be good to ensure we have a test for the actual hashCode we're implementing. At the moment, we are only verifying that the hashCode is the same for two equal UUIDs and different for two unequal UUIDs. One option would be to have a few tests where we verify that the result is what we expect it to be.